edgedb/con_utils.py

```diff
@@ -326,6 +369,76 @@ def _validate_user(user):
     return user
 
 
+def _parse_iso_duration(string: str):
+    match = ISO_DURATION_RE.match(string)
+    if match is None:
+        raise ValueError(f"invalid duration {string!r}")
+
+    return (
+        3600 * float(match.group('hours') or '0') +
+        60 * float(match.group('minutes') or '0') +
+        float(match.group('seconds') or '0')
+    )
+
+
+def _parse_human_duration_unit(re, string):
+    number = None
+    match = re.search(string)
+    if match:
+        number = match.group('number')
+        tail = match.group('tail')
+        string = re.sub(tail, string, count=1)
```

Comment:
What is this `re.sub` for?

Reply:
Repeated units are not valid. 

```
edgedb> select <duration>'1hour 1hour';
edgedb error: InvalidValueError: invalid input syntax for type std::duration: "1hour 1hour"
```

And this enables checking for extra text in the string that is not a unit i.e. `1 hour ago we went to the store...`.

Reply:
changed to `str.replace()`